src/main/scala/io/iohk/ethereum/jsonrpc/ExpiringMap.scala

```diff
@@ -0,0 +1,60 @@
+package io.iohk.ethereum.jsonrpc
+
+import java.time.{Clock, Duration, Instant}
+
+import io.iohk.ethereum.jsonrpc.ExpiringMap.ValueWithDuration
+
+import scala.collection.mutable
+import scala.util.Try
+
+object ExpiringMap {
+
+ case class ValueWithDuration[V](value: V, expiration: Instant)
+
+  def empty[K, V](defaultElementRetentionTime: Duration,
+                  clock: Clock = Clock.systemUTC()): ExpiringMap[K, V] =
+    new ExpiringMap(mutable.Map.empty, clock, defaultElementRetentionTime)
```

Comment:
I know this wasn't introduced in this PR but I'm not sure using this map is the best alternative here as, If i'm not mistaken, our Http server accepts multiple concurrent requests.

Reply:
I wonder if using `Clock` is the way to go. Theoretically an attacker could keep the key unlocked indefinitely by adjusting the system clock.

Reply:
@AlanVerbner yep I pondered on this one a little bit, but eventually, I decided that if the map was good enough till now I will leave it that way. If we are starting to worry about concurrency issues the simplest and most foolproof way would be to wrap this `ExpiringMap` in Actor to ensure proper comunication.  What do you think ? 

@rtkaczyk interesting point, but isn't it that the case, that if an attacker could adjust our system clock we are already so compromised that keeping our key unlocked indefinitely is the least of our problems? I mean `Instant.now(Clock.systemUTC)` under the hood just calls `System.currentTimeMillis()`, and to the attacker to tinker with that, he would have to have full admin rights over the machine. Of course, I am not cybersec specialist so I am maybe missing something. Anyhow, we need some notion of passing time.

Reply:
I'm not sure how much of a security issue this is. The only plausible scenario I can think of is that once someone took control of the machine running Mantis, they can buy themselves some time to figure out what do about unlocked keys (if any).

Anyway, if we could use `System.nanoTime()` here that should make us resilient to any time changes, and that also includes some potentially confusing behaviour in the event of DST or geographical time changes.

As for the `Map`, given how this API is used , this bug is very unlikely to manifest, so maybe a todo comment / separate task?

Reply:
👍 for both suggestions

Reply:
Todo comment works for me